### ISSUE TITLE
Add mgrid/ogrid

### ISSUE THREAD

**Issue by jakirkham** [dask/dask#2939]
Would be nice to have Dask Array implementations of NumPy's [`mgrid`]( https://docs.scipy.org/doc/numpy-1.13.0/reference/generated/numpy.mgrid.html ) and [`ogrid`]( https://docs.scipy.org/doc/numpy-1.13.0/reference/generated/numpy.ogrid.html ). These should be pretty trivial extensions of `meshgrid`. Can also simplify `indices` further using `mgrid`.

**Comment by jakirkham** [comment]
One tricky thing with these is they work by using `__getitem__`. So there isn't a good way to supply a `chunks` argument. If anyone has good ideas for workarounds, would be interested in hearing them.

**Comment by TAdeJong** [comment]
I guess creating them as a class with an `__init__(chunks=(n,m)):` would give a possibility to call them as `da.ogrid(chunks=(1000,1000))[...]` . But I would not know whether this falls under the term "good ideas" :D .

**Comment by jakirkham** [comment]
Yeah was thinking something similar. Honestly think that is an ok approach.

An alternative would be to expose a Dask Array equivalent of `nd_grid` to the public API and allow it to take `chunks`. Though it is a little clumsier than your proposal as one would specify a `sparse` flag too.

**Comment by jakirkham** [comment]
Remembering that `mgrid`/`ogrid` vs. `meshgrid` came up during the recent sprint. Any thoughts on how we should implement these, @mrocklin?

**Comment by mrocklin** [comment]
No particular thoughts from me
